tasks/letsencrypt.yml

```diff
@@ -1,41 +1,51 @@
 ---
+- name: stat certbot-auto /usr/local/bin/ script
+  stat:
+    path: /usr/local/bin/certbot-auto
+  register: file_details
+
 - name: install gnupg2
   apt:
     name: gnupg2
     state: present
     cache_valid_time: 3600
+  when: not file_details.stat.exists
 
 - name: download the certbot-auto signature file
   get_url:
     url: https://dl.eff.org/certbot-auto.asc
     dest: /tmp/certbot-auto.asc
     force: yes
```

Comment:
I'm thinking `force` should be `no` here and in the following state, per these docs: https://docs.ansible.com/ansible/latest/modules/get_url_module.html and then the 2 states after that should be updated to run only if one of these changes. I'd also change this state to save the signature file in `/root` or something so it's persistent across reboots, etc.